File: App/Questionnaire/src/main/java/com/example/mentalhealth/aiService/DTO/RiskSignalDTO.java
package com.example.mentalhealth.aiService.DTO;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RiskSignalDTO {

    private String signalId;           // suicide_risk_flag
    private String level;              // NONE | LOW | MODERATE | HIGH
    private String sourceComponentId;  // phq9_q9
    private String clinicalNote;       // short factual statement
}

File: App/AI-service/models/RiskSignalDTO.py
from pydantic import BaseModel, Field
from typing import Optional


class RiskSignalDTO(BaseModel):
    signalId: str = Field(
        ..., description="Risk signal identifier"
    )

    level: str = Field(
        ..., description="NONE | LOW | MODERATE | HIGH"
    )

    sourceComponentId: Optional[str] = None

    clinicalNote: Optional[str] = Field(
        None,
        description="Short factual note about the signal"
    )

    class Config:
        extra = "forbid"